### ISSUE TITLE
Must home axis first when resume after fillament runout

### ISSUE THREAD

**Issue by lucwuyts** [Bully85/Sovol-SV04-Klipper#26]
Fillament was runout , changed it, and pressed the resume button.
Now this message appears:
Must home axis first: 128.074 62.714 37.094 [142520.420]

Why is this?  The printer clearly knows where the head is.
And homing will collide with the unfinished print

**Comment by Bully85** [comment]
Where do you press resume?

**Comment by Bully85** [comment]
Can you try to reproduce it and make a video?

**Comment by lucwuyts** [comment]
I press resume on the PAD 7 screen.
Will try to make a video.

**Comment by lucwuyts** [comment]
First attempt: print continued as expected.   But this was now at the first layer.    I will try tomorrow with a simple tower at higher Z to repeat this.

**Comment by Bully85** [comment]
Okay, I'm waiting for feedback

**Comment by lucwuyts** [comment]
It seems after 2 hours, the bed and extruder heating stops, and when you press the resume button, this message appears.
Probably this is a bug in the PAD7  or Klipper itself?   
I attach 2 photo's.  The first showing the printer is paused and extruder has cooled down, second after pressing the resume button.
In the 2 hours between the sensor detecting no more fillament, i was always able to restart the print.

![IMG_20231021_133550](https://github.com/Bully85/Sovol-SV04-Klipper/assets/198423/3036864f-7f7b-4f8d-8122-667c1249f928)

![IMG_20231021_133557](https://github.com/Bully85/Sovol-SV04-Klipper/assets/198423/7e063feb-29dc-42c2-8694-13a284ef38c6)

**Comment by lucwuyts** [comment]
I found an example to add to the printer.cfg file about controlling the idle_timeout.
But in this example, the temperature of the extruder is set to 0, the bed temperature is untouched.

When my printer has stopped with his current job, I will try this without the M104 code....  

```
[idle_timeout]
gcode:
{% if printer.pause_resume.is_paused %}
M104 S0
{% else %}
TURN_OFF_HEATERS
M84
{% endif %}

```

**Comment by Bully85** [comment]
M84 disable steper !!!!!!!

**Comment by lucwuyts** [comment]
Yes, I don't care about the disable steppers instruction since the printer is possibly in a canceled state.
This addition solved the problem.  (I did erase the line with M104   S0)
I have been away for more than 4 hours, just returned, inserted the fillament again, and pushed resume.   Started as it should.
